ocs-api/data/data.go

```diff
@@ -98,20 +98,38 @@ var SviJson3 = `_exec",
   }
 `
 
-var AgentInstallWrapper = `#!/bin/bash
+var AgentInstallWrapper = `#!/bin/sh
 
 # The primary purpose of this wrapper is to be able to invoke agent-install.sh in the SDO context and log all of its stdout/stderr
 
-logFile=/tmp/agent-install.log
-echo "Logging all output to $logFile"
-
 # Verify the number of args is what we are handling below
 numRequiredArgs=8
-if [[ $# -ne $numRequiredArgs ]]; then
+if [ $# -ne $numRequiredArgs ]; then
     echo "Error: expected $numRequiredArgs arguments and received $#"
     exit 2
 fi
 
+echo "$0 starting..."
+
+# This script has a 2nd purpose in the native client case: when run inside the docker sdo container, copy the downloaded files to outside the container
+if [ -f /target/boot/inside-sdo-container ]; then
+    # Copy all of the downloaded files (including ourselves) to /target/boot, which is mounted from host /var/horizon/sdo-native
+    echo "Copying downloaded files to /target/boot: $(ls | tr "\n" " ")"
+    # need to exclude a few files and dirs, so copy with find
+    find . -maxdepth 1 -type f ! -name inside-sdo-container ! -name linux-client ! -name run_csdk_sdo.sh -exec cp -p -t /target/boot/ {} +
+    if [ $? -ne 0 ]; then echo "Error: can not copy downloaded files to /target/boot"; fi
```

Comment:
This whole `if` block would never be run on a Mac would it? I only ask because I just ran `man cp` and realized there is no `-t` option for Mac apparently

Reply:
Correct, that block of code only runs inside the `sdo:1.8` container (so always linux) which the Intel team built to make it easier to run the native (C code) SDO client on physical Intel and arm servers.

Related info: i've made the architectural decision that we won't support a mac as an sdo-enabled device. The code is complicated enough w/o trying to support a non-linux OS like that. (I **do** still want to support running the sdo-owner-services container on mac if it doesn't cause us any problems.)